lib/kb/fake/bounded_context/pet_family/pet_contracts.rb

```diff
@@ -17,8 +20,23 @@ module PetContracts
           json_response 200, resource
         end
 
+        get '/v1/petcontracts/search' do
+          pet_by_chip = resource_state(:pets).detect do |pet|
+            pet['chip'] == params['chip']
+          end
+          return json_response 404, {} if pet_by_chip.nil?
+
+          contracts = resource_state(:petcontracts).select do |contract|
+            contract['petKey'] == pet_by_chip['key']
+          end
+          return json_response 404, {} if contracts.nil?
```

Comment:
And same here. It's supposed to be 200 with empty response. Or does API really returns 404?